src/main/java/ch/njol/skript/events/EvtGrow.java

```diff
@@ -55,48 +53,38 @@ public class EvtGrow extends SkriptEvent {
 	private Literal<StructureType> types;
 	@Nullable
 	private Literal<ItemType> blocks;
-	private int evtType;
+	private int eventType;
 	
 	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(final Literal<?>[] args, final int matchedPattern, final ParseResult parser) {
-		evtType = parser.mark; // ANY, STRUCTURE or BLOCK
-		if (evtType == STRUCTURE)
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		eventType = parseResult.mark; // ANY, STRUCTURE or BLOCK
+		if (eventType == STRUCTURE) {
 			types = (Literal<StructureType>) args[0];
-		else if (evtType == BLOCK)
+		} else if (eventType == BLOCK) {
 			blocks = (Literal<ItemType>) args[1]; // Arg 1 may not be present... but it is in the array still, as null
+		}
 		// Else: no type restrictions specified
 		return true;
 	}
 	
 	@Override
-	public boolean check(final Event e) {
-		if (evtType == STRUCTURE  && types != null && e instanceof StructureGrowEvent) {
-			return types.check(e, new Checker<StructureType>() {
-				@Override
-				public boolean check(final StructureType t) {
-					return t.is(((StructureGrowEvent) e).getSpecies());
-				}
-			});
-		} else if (evtType == BLOCK && blocks != null && e instanceof BlockGrowEvent) {
-			assert blocks != null;
-			return blocks.check(e, new Checker<ItemType>() {
-				@Override
-				public boolean check(final ItemType t) {
-					return t.isOfType(((BlockGrowEvent) e).getBlock());
-				}
-			});
+	public boolean check(Event event) {
+		if (eventType == STRUCTURE  && types != null && event instanceof StructureGrowEvent) {
+			return types.check(event, type -> type.is(((StructureGrowEvent) event).getSpecies()));
+		} else if (eventType == BLOCK && blocks != null && event instanceof BlockGrowEvent) {
+			return blocks.check(event, type -> type.isOfType(((BlockGrowEvent) event).getNewState()));
```

Comment:
Actually I forgot to comment on this, this doesn't make much sense, when you're waiting for something to grow you're expecting to give the old state of the growing thing not the new one.

I haven't made any tests yet on your example and other examples so I will try to test it and give more feedback.

Reply:
Hmm. Maybe 'grow from' and 'grow to'? A little cumbersome but clear.

Reply:
Indeed it's clear and a good solution IMO, the syntax might be odd but it's fine to me.
I would discuss this more with the team in skriptlang discord channel on skUnity to get more feedback.

My feedback is:
- The proposed syntax you gave `from` and `to` is fine to me
- The default behavior should be reverted to avoid breaking changes (unstable decision tbh due to your example)

Reply:
The source block expression could be used.

Reply:
first:
I'm trying to think of how to word this and it's really weird.
``` 
on grow from sapling:          # nice and good
on grow to tree:               # pretty awkward
on grow of tree:               # makes sense, but can be confused (sapling -> tree, or tree -> ??)

on grow from wheat:            # these are kind of redundant when not specifying an age
on grow to wheat:              # from and to are a little awkward as well
on grow of wheat:                      

on grow from wheat[age=4]:     # alright i get it
on grow to wheat[age=4]:       # still awkward
on grow of wheat[age=4]:       # sounds great, but again is it age 3 -> 4 or age 4 -> 5)

on grow from wheat[age=4] to wheat[age=5]:       # very clear and good but quite specific and annoying to write

on grow from pumpkin:          # feels natural to use but will never be called. instead on grow from air will be. (and from stem won't work either)
on grow to pumpkin:            # same as last few times, awkward but ok
on grow of pumpkin:            # again, same confusion
```
What I really want to do is use `on grow[th] of X` as from something into X, which we already talked about and acknowledged will cause confusion. However, I feel it's the most natural because we already use `on grow of tree`, where tree is the end result, not the starting thing (a sapling).
Then we'd just have `on grow[th] of` and, i guess, `on grow[th] from`? which also isn't great. 
I don't really know, I feel like there are no good options. Gonna ask for more opinions on discord too.

second:
I don't think I can easily pull in the source block expression, since BlockSpread is a child of BlockGrow, and it's also kinda using a different method already? If I had it act as `past event-block` for BlockGrow, it'd be weird for it to not be that for BlockSpread. And it really should give the stem in a melon/pumpkin growth event, but that's not accessible from the BlockGrow event. So it seems not worth the attempt. If anyone has solutions to this i'd love to hear them tho.

Reply:
```
# x -> something
on grow[th] from X

# something -> x
on grow[th] into X

# X -> Y
on grow[th] from X [in]to Y

# x is involved in some way
on grow[th] of X
```
Conclusion

Reply:
Sounds good. Will we still have to provide `from air` for pumpkin?

Reply:
Now we are talking 😎 sounds very good results to me. Well done Sovde 👏